### PR TITLE
Refactor and update regulation layers color

### LINKED ISSUE #1420: Changement des couleurs des couches REG et affichage des périodes non autorisées
Beaucoup de zones avec engins n'ont pas de couleur correspondante. Et les couleurs des zones ne sont pas reportées dans la vue liste (petit carré à gauche du nom de la zone toujours de la couleur par défaut).

### PR DESCRIPTION
**Pull request by louptheron**
## Linked issues

- Resolve #1420

----

- [ ] Tests E2E (Cypress)

### CODE REVIEW

File: frontend/src/ui/theme.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
```
Comment (ivangabriele) [inline frontend/src/ui/theme.ts:1]: We will have a massive conflict on this file with [my color PR ](https://github.com/MTES-MCT/monitorfish/pull/1437). Can we put these colors in another props `colorsTemp: {}` while the cleaning is not finished?

**Review by ivangabriele — APPROVED** [review]

File: frontend/src/domain/types/layer.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export type AdministrativeOrRegulatoryLayer = {
+export type ShowAdministrativeOrRegulatoryLayer = {
```
[the remark this answers is not in the document]
Comment (louptheron) [reply frontend/src/domain/types/layer.ts:1]: I agree, let's use `AdministrativeOrRegulatoryLayerIdentity`, as this is used for showing, hiding layers and it contains only the identity part of the layer.

File: frontend/src/layers/styles/unAuthorizedPeriod.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    default: {
+      return false
+    }
+  }
+}
+
+/**
+ * It is unauthorized when the date is out of at least one fishing period.
+ */
+function getIsUnAuthorizedForAuthorizedPeriod(fishingPeriod: FishingPeriod, now: Dayjs, currentWeekDayDigit: number) {
```
[the remark this answers is not in the document]
Comment (louptheron) [reply frontend/src/layers/styles/unAuthorizedPeriod.ts:34]: I prefer `currentDate`

File: frontend/src/layers/styles/utils.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      color: 'rgba(5, 5, 94, 0.7)',
+      width: isSelected ? 3 : 1
+    })
+  })
+}
+
+/**
+ * Given a color in hex format and a alpha number, returns the color as an rgba string.
+ */
+export function getColorWithAlpha(colorHex: string, alpha: number) {
```
[the remark this answers is not in the document]
Comment (louptheron) [reply frontend/src/layers/styles/utils.ts:21]: Why the root folder ? This util is only used in this `styles/` directory

**Pull request merged**